app/reviews/models.py

```diff
@@ -25,16 +27,36 @@ class Review(TimestampMixin):
     )
 
     # Make nullable for anonymity of reviewers.
-    user = models.ForeignKey(User, on_delete=models.SET_NULL, blank=True, null=True)
-    # anonymity_id is just a unique identifier for the reviewer used
+    user = models.ForeignKey(
+        User, on_delete=models.SET_NULL, blank=True, null=True
+    )
+    # anonymity_id is just a hashed unique identifier for the reviewer used
     # as a check prevent multiple reviews from one person
     anonymity_id = models.CharField(blank=True, null=True, max_length=250)
     serving = models.ForeignKey(Serving, on_delete=models.CASCADE)
     value = models.PositiveSmallIntegerField(choices=RATINGS)
     comment = models.TextField(blank=True)
 
     def __str__(self):
-        return '{}\'s review of {}'.format(self.user, self.serving)
-
-    class Meta:
-        unique_together = ('user', 'serving')
+        return '{}\'s review of {}'.format(
+            self.user or self.anonymity_id, self.serving
+        )
+
+    def clean(self):
+        if self.user:
```

Comment:
With this, I think multiple reviews from one user (with their identity not specified) is possible. Same with the `anonymity_id`, a case where the client does not send it along makes multiple reviews from a single reviewer possible as we decided to make it flexible for clients to decide whether to send `anonymity_id` or not. Any idea on how to prevent this? Or is it taken care of somewhere else already? Or am I mixing things up?